src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -86,12 +85,59 @@ public ModelAndView classSetup(@PathVariable String sectionId,
         return page;
     }
 
+    private ModelAndView validateInputType(String sectionId, CourseConfigurationForm classSetupForm) {
+        if (classSetupForm.getGradingOn()){
+            boolean typeValid = (classSetupForm.getAssignmentPoints().matches("[0-9.]+") || classSetupForm.getAssignmentPoints() == null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (9) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1067)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "[0-9.]+" 10 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)